crates/pera-core/src/dwallet_mpc/mpc_manager.rs

```diff
@@ -240,55 +298,53 @@ impl DWalletMPCManager {
             .mpc_sessions
             .iter_mut()
             .filter_map(|(_, session)| {
-                let received_weight: PartyID =
-                    if let MPCSessionStatus::Active(round) = session.status {
-                        session.pending_messages[round]
-                            .keys()
-                            .map(|authority_index| {
-                                // Should never be "or"
-                                // as we receive messages only from known authorities.
-                                self.weighted_threshold_access_structure
-                                    .party_to_weight
-                                    .get(authority_index)
-                                    .unwrap_or(&0)
-                            })
-                            .sum()
-                    } else {
-                        0
-                    };
-
-                let is_ready = (matches!(session.status, MPCSessionStatus::Active(_))
-                    && received_weight as StakeUnit >= threshold)
-                    || (session.status == MPCSessionStatus::FirstExecution);
-
-                let mut is_manager_ready = true;
-                if cfg!(feature = "with-network-dkg") {
-                    is_manager_ready = (mpc_network_key_status
-                        == DwalletMPCNetworkKeysStatus::NotInitialized
-                        && matches!(session.party(), MPCParty::NetworkDkg(_)))
+                let received_weight: PartyID = match session.status {
+                    MPCSessionStatus::Active => session
+                        .pending_messages
+                        .get(session.round_number)
+                        .unwrap_or(&HashMap::new())
+                        .keys()
+                        .filter_map(|authority_index| {
+                            self.weighted_threshold_access_structure
+                                .party_to_weight
+                                .get(authority_index)
+                        })
+                        .sum(),
+                    _ => 0,
+                };
+
+                let is_ready = match session.status {
+                    MPCSessionStatus::Active => received_weight as StakeUnit >= threshold,
+                    MPCSessionStatus::FirstExecution => true,
+                    _ => false,
+                };
+
+                let is_valid_network_dkg_transaction =
+                    matches!(session.session_info.mpc_round, MPCRound::NetworkDkg(..))
+                        && self.validators_data_for_network_dkg.len()
+                            == self
+                                .weighted_threshold_access_structure
+                                .party_to_weight
+                                .len();
```

Comment:
We don't want validators to be able to DOS by preventing a DKG from happening by simply not supplying you with their `data_for_network_dkg`. Instead, you should pad it with default values for every validator that didn't send you this data; my code has validity checks, and will simply report that validator as malicious.

Reply:
Alternatively, we can change my code to be able to get a partial mapping, or have the values as `Option` and report every missing player as malicious.